### PR TITLE
 Columns don't provide a WYSIWYG experience in edit mode #208 

### PR DESCRIPTION
**Pull request by Farcasut**
I've modified the structure of the cke wrappers using CSS and JavaScript to make the page look as expected. To be more precise, I've made the main wrapper a flex container to display each column on the same line and moved the width from the original element to its wrapper to keep the ratio used in the view mode.

### CODE REVIEW

**Comment by Farcasut** [comment]
image

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml [hunk 1]
```diff
@@ -241,6 +352,19 @@ Content in the column 4
     <property>
       <code>.hasBorder .macro-column {
   border: 1px dashed grey;
+}
+.macro-section
+{
+  display:flex;
+}
+.macro-section {
```
Comment (trrenty) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml:360]: Duplicate code

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml [hunk 2]
```diff
@@ -134,6 +134,117 @@ Content in the column 4
...
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery'], ($) =&gt; {
```
Comment (trrenty) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml:225]: you should also require this jsx code in the section macro code

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml [hunk 3]
```diff
@@ -134,6 +134,117 @@ Content in the column 4
...
+});</code>
+    </property>
+    <property>
+      <name/>
+    </property>
+    <property>
+      <parse/>
+    </property>
+    <property>
+      <use/>
```
Comment (trrenty) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Section.xml:245]: onDemand. Im not sure whats the default, but it doesnt hurt to have it specified.

**Review by trrenty — CHANGES_REQUESTED** [review]
Some changes needed

**Review by trrenty — APPROVED** [review]

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Column.xml [hunk 4]
```diff
@@ -437,7 +437,7 @@ Content in the column 4
   #if ("$!xcontext.macro.params.width" != '')
     #set ($style = "style='width: $xcontext.macro.params.width'")
   #end
-  (% class="macro-column" $style %)(((
+  (% class="macro-column" data-width="$xcontext.macro.params.width" $style %)(((
```
Comment (mflorea) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Column.xml:440]: Missing escaping. 3 lines above too.

**Pull request merged**